File: Java/Singleton/Singleton.java
public class Singleton {
    public static Singleton instance = null;

    public int counter = 0;

    private String message = "";

    private Singleton() {}

    public static Singleton getInstance() {
        if (instance == null) {
            instance = new Singleton();
        }
        return instance;
    }

    public void increment() {
        counter++;
    }
    public void decrement() { counter--; }

    public String getMessage() {
        return message;
    }
    public void setMessage(String message) {
        this.message = message;
    }
}

File: JavaScript/Singleton/Singleton.ts
class Singleton {
    static #instance:Singleton = null;
    #counter: number = 0;
    #message: string = "";

    constructor() {}

    static getInstance() {
        if (this.#instance == null) {
            this.#instance = new Singleton();
        }
        return this.#instance;
    }

    getCounter(): number {
        return this.#counter;
    }
    increment() : void {
        this.#counter++;
    }
    decrement() : void {
        this.#counter--;
    }

    setMessage(message: string) : void {
        this.#message = message;
    }
    getMessage(): string {
        return this.#message;
    }
}

const instance1: Singleton = Singleton.getInstance();
const instance2: Singleton = Singleton.getInstance();

instance1.increment()
instance2.setMessage("Hello Jupiter")
console.log(instance2.getCounter())
console.log(instance1.getMessage())
